add_full_index_hash.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Nov 29 09:57:10 2017
+Small script to add hashes of indices to the DHS PSQL tables.
+Intended to be run on the PSQL machine.
+@author: Craig
+"""
+
+# The DHS tables, as inserted into our PSQL DB, are "chunked", that is, split
+# by width since they have too many columns. So, e.g., we will have one table
+# "DHS-Guatemala-Children Recode-v71", and then 5 more tables "D-G-CR-v71-N",
+# N=1,...,5. Each of these tables has a set of columns serving as a common
+# index, but attempting to join said tables in an automated fashion requires
+# having a map from baseable to set-of-index-columns.
+# Instead, we are going to add a new column, "full_index_hash", containing
+# the hash of the index column values. This way, when we want to join, we can
+# join on that one column alone and don't need the map.
+
+import re
+from sqlalchemy import create_engine
+
+# Index columns are identified by having one of these substrings.
+INDEX_COLUMNS = [ "Facility number", "Unit line number", "unit type",
+    "provider line number", "Case Identification", "Cluster number",
+    "Household number", "Respondent\'s line number", "Country code" ]
+
+
+def get_index_column_query(table_name):
+  query = "SELECT column_name FROM information_schema.columns "
+  query += "WHERE table_name = \"" +table_name+ "\" AND (column_name ILIKE \"%"
+  query += "%\" OR column_name ILIKE \"%".join(INDEX_COLUMNS)
+  query += "%\")"
+  return query
+  
+
+def main():
+  pg_username = input("Please enter Postgres username:")
+  pg_password = input("Password:")
```

Comment:
Please use a function that will mask the password.

Reply:
This is intended to be run locally on the AIGH machine.